pkg/attacher/attacher.go

```diff
@@ -88,11 +85,22 @@ func (a *attacher) Detach(ctx context.Context, volumeID string, nodeID string, s
 		Secrets:  secrets,
 	}
 
-	_, err = client.ControllerUnpublishVolume(ctx, &req)
+	_, err := client.ControllerUnpublishVolume(ctx, &req)
 	if err != nil {
-		return isFinalError(err), err
+		// Consume NotFound error.
+		st, ok := status.FromError(err)
+		if !ok {
+			// This is not gRPC error.
+			return err
+		}
+		if st.Code() == codes.NotFound {
```

Comment:
what about a case where `SP` temporarily cannot "find" the volume or the disk, but the disk is still attached. This should be `NOT_FOUND` error that could be remedied by retrying - disk is not actually detached yet.

I would argue that the `SP` should return `OK` if it has determined that the volume/node is gone in a way that we can assume the volume is detached. This should be up to the `SP` to decide.

Reply:
We're assuming all volume plugins automatically detach volumes when a node is deleted, and I'm not confident that we can assume that.

Reply:
I think it is a reasonable compromise until this behaviour is clarified in spec. I do not know any volume type where attachment outlives lifecycle of node.

Reply:
Sorry ignore my last comment, I misread SP as CO. We're on the same page :-)

Reply:
Either way I think we are asking plugin authors to reinterpret spec in a way which was previously not documented. 

Esp. for node deletion case - do we have a solution in the meanwhile?

Reply:
I think vsphere may have some interesting behaviors around node failure/deletion handling. cc @codenrhoden @vladimirvivien

Reply:
afaik - in vsphere too, when you delete a node in vcenter, the vmdk files are marked as detached.